src/lib/useEthereumTransaction.js

```diff
@@ -135,40 +143,46 @@ export default function useEthereumTransaction(
       setConfirmationProgress(0.0);
       setError(undefined);
 
-      const rawTx = hexify(signedTransaction.serialize());
-      const costGwei = toBN(gasLimit).mul(toBN(gasPrice));
-      const cost = toWei(costGwei.toString(), 'gwei');
-
-      let usedTank = false;
-      // if this ethereum transaction is being executed by a specific point
-      // see if we can use the tank
-      if (Just.hasInstance(pointCursor)) {
-        usedTank = await ensureFundsFor(
-          _web3,
-          pointCursor.value,
-          _wallet.address,
-          cost,
-          [rawTx],
-          (address, minBalance, balance) =>
-            setNeedFunds({ address, minBalance, balance }),
-          () => setNeedFunds(undefined)
-        );
-      }
-
-      const txHash = await sendSignedTransaction(
-        _web3,
-        signedTransaction,
-        /* doubtNonceError */ usedTank
+      const txHashes = await Promise.all(
+        signedTransaction.map(async stx => {
+          const rawTx = hexify(stx.serialize());
+          const costGwei = toBN(gasLimit).mul(toBN(gasPrice));
+          const cost = toWei(costGwei.toString(), 'gwei');
+
+          let usedTank = false;
+          // if this ethereum transaction is being executed by a specific point
+          // see if we can use the tank
+          if (Just.hasInstance(pointCursor)) {
+            usedTank = await ensureFundsFor(
+              _web3,
+              pointCursor.value,
+              _wallet.address,
+              cost,
+              [rawTx],
+              (address, minBalance, balance) =>
+                setNeedFunds({ address, minBalance, balance }),
```

Comment:
This is going to be inaccurate for a couple reasons:
- It's only going to show the requirements for a single transaction, even if there's many more before/after it.
- It doesn't account for funds used by other transactions, assuming that all of these `unsureFundsFor` calls happen more or less in parallel.

I think there's two sane options:
- `ensureFundsFor` the max total of all transactions combined. This'll overshoot the real cost/requirements, but at least you're only asking the user once at most.
- Process transactions in serial. You'll still have the "ask user multiple times" problem, but you can reuse funds left untouched by previous transactions, and can stop sending the rest if one tx fails (better error handling). This will, of course, be way slower though.

The former makes error handling a bit messy, but _should_ be fine for the star unlocking case. Perhaps we want to implement both (`submitTxsParallel` and `submitTxsSerial`), and decide which to use depending on the context?

Reply:
Can you clarify exactly how the first one will overshoot? I'm not really an ethereum expert

Reply:
Decided on a compromise of:

1. `ensureFundsFor` the max total. I don't think it's a great UX to be continually pestered for more eth.
2. Transactions sent in serial. Agreed re: error handling.

Just pushed, what do you think?

Reply:
> Can you clarify exactly how the first one will overshoot?

The funds _required_ to _submit_ a transaction are `gas-price * gas-limit`, because it's possible that you'll use as much gas as the `gas-limit`.  
In most cases (clean & successful transactions), you end up _using_ less gas than the limit in computation. Because _you only pay for what you use_, you'll often pay less than the `gas-price * gas-limit`.

So, if you sum the `gas-price * gas-limit` for all transactions, you have the upper bound, but will in practice both require & pay less than that.